File: src/network/LobbyEntry.java
package network;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
public class LobbyEntry {
    public static final int ROOMLESS = -1;

    @Getter
    private int userId;

    private String username;

    @Getter
    @Setter
    private boolean readyForGame;
    @Getter
    @Setter
    private int chosenTeamId;

    LobbyEntry() {
        userId = -1;
    }

    @Override
    public String toString() {
        return String.format("#%d %s (%s)",
                this.userId,
                this.username,
                this.readyForGame ? "ready" : "waiting");
    }
}

File: src/main/kotlin/network/LobbyEntry.kt
package network

class LobbyEntry(
        var userId: Int,
        var username: String?,
        var isReadyForGame: Boolean,
        var chosenTeamId: Int) {

    @Suppress("unused")
    // Kryonet deserialization
    internal constructor() :
            this(0, null, false, -1)

    override fun toString(): String {
        return String.format("#%d %s (%s)", userId, username, if (isReadyForGame) "ready" else "waiting")
    }

    companion object {
        const val ROOMLESS = -1
    }
}
